rewrite-java-test/src/test/java/org/openrewrite/java/ChangeMethodNameTest.java

```diff
@@ -422,4 +428,70 @@ public void test() {
           )
         );
     }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"123", "  ", "\t", "this", "class", "null", "true"})
+    void ignoreInvalidMethodNamesWhenCalledDownstream(String newMethodName) {
+        AdHocRecipe downstreamRecipe = new AdHocRecipe("DownStream", "downstream", false, () -> new JavaIsoVisitor<>() {
+            @Override
+            public J visit(Tree tree, ExecutionContext executionContext) {
+                return (J) new ChangeMethodName("com.abc.B singleArg(String)", newMethodName, null, null).getVisitor().visitNonNull(tree, executionContext);
+            }
+        }, List.of(), 1);
+
+        rewriteRun(
+          spec -> spec.recipe(downstreamRecipe),
```

Comment:
This could be shorter by using `toRecipe` as a static import from `RewriteTest`.
```suggestion
        rewriteRun(
          spec -> spec.recipe(toRecipe(() -> new JavaIsoVisitor<ExecutionContext>() {
              @Override
              public J visit(Tree tree, ExecutionContext executionContext) {
                  return (J) new ChangeMethodName("com.abc.B static1(String)", newMethodName, null, null).getVisitor().visitNonNull(tree, executionContext);
              }
          })),
```

Also; your using `singleArg` in the matcher, but the text block below uses `static1`, so even with a valid name your variant made no match or change. That's also fixed with the above.

Reply:
The toRecipe did not behave the same as the explicit adhoc recipe

Reply:
Thnx! the first version of toRecipe did not show the same behavior as the adhoc recipe, but this one does :)